File: unit2/hello_name.c
#include <cs50.h>
#include <stdio.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./hello_name <first name>\n");
        return 1;
    }

    printf("hello, %s!\n", argv[1]);

    return 0;
}

File: unit3/hello_name.py
# import sys
from sys import argv, exit

def main():
    argc = len(argv)

    if argc != 2:
        print("Usage: ./hello_name <first name>")
        exit(1)

    print(f"hello, {argv[1]}!")

if __name__ == "__main__":
    main()